addons/api/mirage/factories/host.js

```diff
@@ -10,4 +10,9 @@ export default factory.extend({
       'delete',
     ],
   id: (i) => `host-id-${i}`,
+  plugin() {
+    if (this.type === 'plugin') {
+      return this.hostCatalog.plugin;
```

Comment:
just curious, why are we returning `hostCatalog.plugin` instead of `host.plugin`

Reply:
Because we want the object `plugin` inherit from hostCatalog.

Reply:
@DhariniJeeva the type is inherited from the host catalog.  However this raises a problem with the implementation.  We need to apply this inheritance logic to `type` as well @calcaide.

Reply:
@randallmorey 
On it's way! I did it in `host-set` [here](https://github.com/hashicorp/boundary-ui/pull/925/files#diff-c4d10025ef7bd2a9d9275ee8670da51e3d034c096ac9fe931dc4233ea88544b0R8) I will reproduce that pattern here.